uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/sys/PatternFlyBootstrapper.java

```diff
@@ -119,6 +124,28 @@ public static void ensureJQueryUIIsAvailable() {
         }
     }
 
+    public static void ensureMonacoEditorLoaderIsAvailable() {
+        if (!isMonacoEditorLoaderLoaded) {
+
+            final String monacoAbsolutePath = GWT.getModuleBaseURL() + MONACO_EDITOR_BASE_PATH;
+            final String amdLoaderScript = PatternFlyClientBundle.INSTANCE.monacoAMDLoader().getText();
+
+            ScriptInjector.fromString(enclosureByMonacoAMDLoaderNamespace(monacoAbsolutePath, amdLoaderScript))
+                    .setWindow(ScriptInjector.TOP_WINDOW)
+                    .inject();
+
+            isMonacoEditorLoaderLoaded = true;
+        }
+    }
+
+    private static String enclosureByMonacoAMDLoaderNamespace(final String baseUrlPath,
+                                                              final String script) {
+
+        return "(new function() { this.require = { baseUrl: '" + baseUrlPath + "' }; "
+                + script
+                + " window.__MONACO_AMD_LOADER__ = _amdLoaderGlobal });";
```

Comment:
Ouch, there seems to be some hoops you're jumping through to load Monaco..

Is it possible for you to add some JavaDocs [here](https://github.com/kiegroup/appformer/pull/836/files#diff-c5bafbc2a3bfb8f52eaf4f48b434adc1R141) and [here](https://github.com/kiegroup/appformer/pull/836/files#diff-046c0ce38144a43a8111f56cd02d096aR27) explaining the reasons for these _hacks_?

Reply:
I just added the Java Docs. These hacks are needed because the global scope is polluted by other libraries. **Ace** has its own AMD Loader, **Monaco** has its own AMD Loader as well, and there are other libraries relying on functions and objects with generic names in the global scope.

I've tried to update **Ace** with a new version [1] that relies on **RequireJS** as the AMD Loader, so **Monaco** and **Ace** could  use the same loader since both are compatible with **RequireJS**. However, we have other jQuery libraries that rely on functions that invalidate the setup of **RequireJS**. Thus, I gave up on this approach.

Given this scenario, to avoid as many regressions as possible, I'm enclosing **Monaco** instantiation into this temporary scope, and exposing its loader into this namespace: `__MONACO_AMD_LOADER__` - the only global namespace that **Monaco** will use. However, when we require **Monaco** modules, it needs to temporary use the global namespace for calling `define` and `require` functions, that's why I've encapsulated this switch-logic into `MonacoEditorInitializer`.

[1] Another problem with this approach is the number of regressions related to old Ace APIs.